env/fs_posix.cc

```diff
@@ -1028,11 +1013,25 @@ class PosixFileSystem : public FileSystem {
         return false;
     }
 #else
-    (void)path;
+    (void)fd;
     return false;
 #endif
   }
 
+  void MaybeForceDisableMmap(int fd) {
+    static std::once_flag s_check_disk_for_mmap_once;
+    std::call_once(
+        s_check_disk_for_mmap_once,
+        [this](int fdesc) {
+          // this will be executed once in the program's lifetime.
```

Comment:
I would feel slightly better if there was `assert(this == FileSystem::Default().get());` to ensure singleton.